python/pdo/test/contract.py

```diff
@@ -298,12 +324,19 @@ def CreateAndRegisterContract(config, enclave, contract_creator_keys) :
 
 # -----------------------------------------------------------------
 # -----------------------------------------------------------------
-def UpdateTheContract(config, enclave, contract, contract_invoker_keys) :
-    txn_dependencies = []
+def UpdateTheContract(config, contract, enclaves, contract_invoker_keys) :
+    commit_dependenices = []
 
     ledger_config = config.get('Sawtooth')
     contract_invoker_id = contract_invoker_keys.identity
 
+    # Decide if the contract use a fixed enclave or a randomized one for each update. If fixed, we chose here. If random, nothing to be done here, 
+    # will be (atuomatically) selected at random during request creation 
+    if (use_eservice is False) or (config['Service']['Randomize_Eservice'] is False):
+        enclave_to_use = enclaves[0]
+    else:
+        enclave_to_use = None
+
```

Comment:
as we've discussed, i think this is overly complex & doesn't use the abstractions that we have